spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/mapping/FirestoreDefaultClassMapper.java

```diff
@@ -54,14 +57,37 @@ public FirestoreDefaultClassMapper() {
 
 	public <T> Document entityToDocument(T entity, String documentResourceName) {
 		DocumentSnapshot documentSnapshot = INTERNAL.snapshotFromObject(NOT_USED_PATH, entity);
-		Map<String, Value> valuesMap = INTERNAL.protoFromSnapshot(documentSnapshot);
 		return Document.newBuilder()
-				.putAllFields(valuesMap)
+				.putAllFields(removeUpdateTimestamp(INTERNAL.protoFromSnapshot(documentSnapshot), entity))
 				.setName(documentResourceName).build();
 	}
 
 	public <T> T documentToEntity(Document document, Class<T> clazz) {
 		DocumentSnapshot documentSnapshot = INTERNAL.snapshotFromProto(Timestamp.now(), document);
-		return documentSnapshot.toObject(clazz);
+		T entity = documentSnapshot.toObject(clazz);
+		return setUpdateTime(entity, documentSnapshot.getUpdateTime());
+	}
+
+	public  <T> T setUpdateTime(T entity, Timestamp updateTime) {
+		FirestorePersistentEntity<?> persistentEntity =
+				this.mappingContext.getPersistentEntity(entity.getClass());
+		FirestorePersistentProperty versionProperty = persistentEntity.getUpdateTimeProperty();
```

Comment:
`versionProperty` may not be the best name for it, since some people may be using it outside of optimistic locking.